### PR TITLE
Get articles refactor

### PR DESCRIPTION
**Pull request by JoeDGit**

### CODE REVIEW

File: __tests__/endpoint-requests.test.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
+const testData = require('../db/data/test-data');
 
 afterAll(() => db.end());
 
 beforeEach(() => seed(testData));
 
-describe("General API Errors", () => {
-  test("status: 404, should return a 404 error when a user tries to access an invalid path", () => {
+describe('General API Errors', () => {
+  test('status: 404, should return a 404 error when a user tries to access an invalid path', () => {
```
Comment (roisinoreilly) [inline __tests__/endpoint-requests.test.js:12]: your test descriptions are nice and clear 👍

File: __tests__/endpoint-requests.test.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
     return request(app)
-      .get("/api/badpath")
+      .get('/api/badpath')
       .expect(404)
       .then((response) => {
-        expect(response.body.msg).toEqual("path not found");
+        expect(response.body.msg).toEqual('path not found');
```
Comment (roisinoreilly) [inline __tests__/endpoint-requests.test.js:17]: minor point but you could use .toBe here

File: __tests__/endpoint-requests.test.js [hunk 3]
```diff
@@ -63,23 +63,116 @@ describe("GET /api/articles", () => {
...
+    return request(app)
+      .get('/api/articles?sort_by=author&order=asc')
+      .expect(200)
+      .then((response) => {
+        const articles = response.body.articles;
+        expect(articles).toBeSortedBy(articles.author);
+      });
+  });
+
+  test('status: 200, should allow for mixed case queries', () => {
```
Comment (roisinoreilly) [inline __tests__/endpoint-requests.test.js:132]: nice edge case

File: __tests__/endpoint-requests.test.js [hunk 4]
```diff
@@ -63,23 +63,116 @@ describe("GET /api/articles", () => {
...
+    return request(app)
+      .get('/api/articles?topic=cats&sort_by=author&order=up')
+      .expect(400)
+      .then((response) => {
+        const error = response.body.msg;
+        expect(error).toBe('bad request - invalid sort order');
+      });
+  });
+});
+
```
Comment (roisinoreilly) [inline __tests__/endpoint-requests.test.js:171]: really great testing. have a think about testing for if the query is valid but there are no results

File: models/models.articles.js [hunk 5]
```diff
@@ -1,14 +1,53 @@
...
+    if (topic) {
+      SQL += ` WHERE topic = $1`;
+      queryValues.push(topic);
+    }
+
+    SQL += ` GROUP BY articles.article_id
+ORDER BY ${sort_by} ${order}`;
+
+    return db.query(SQL, queryValues).then((articles) => {
+      return articles.rows;
```
Comment (roisinoreilly) [inline models/models.articles.js:49]: very nicely done

**Review by roisinoreilly — APPROVED** [review]
excellent work, PR approved merge when ready
